java/semantickernel-extensions-parent/sequentialplanner-extensions/src/main/java/com/microsoft/semantickernel/planner/sequentialplanner/SequentialPlanParser.java

```diff
@@ -0,0 +1,200 @@
+// Copyright (c) Microsoft. All rights reserved.
+package com.microsoft.semantickernel.planner.sequentialplanner;
+
+import com.microsoft.semantickernel.builders.SKBuilders;
+import com.microsoft.semantickernel.orchestration.SKContext;
+import com.microsoft.semantickernel.orchestration.SKFunction;
+import com.microsoft.semantickernel.orchestration.WritableContextVariables;
+import com.microsoft.semantickernel.planner.PlanningException;
+import com.microsoft.semantickernel.planner.actionplanner.Plan;
+import com.microsoft.semantickernel.skilldefinition.FunctionView;
+import com.microsoft.semantickernel.skilldefinition.ReadOnlySkillCollection;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.w3c.dom.Document;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+import org.xml.sax.SAXException;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+import java.util.Objects;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+
+/** Parse sequential plan text into a plan. */
+public class SequentialPlanParser {
+    private static final Logger LOGGER = LoggerFactory.getLogger(SequentialPlanParser.class);
+
+    // The tag name used in the plan xml for the user's goal/ask.
+    private static final String GoalTag = "goal";
+
+    // The tag name used in the plan xml for the solution.
+    private static final String SolutionTag = "plan";
+
+    // The tag name used in the plan xml for a step that calls a skill function.
+    private static final String FunctionTag = "function.";
+
+    // The attribute tag used in the plan xml for setting the context variable name to set the
+    // output of a function to.
+    private static final String SetContextVariableTag =
+            "setContextVariable".toLowerCase(Locale.ROOT);
+
+    // The attribute tag used in the plan xml for appending the output of a function to the final
+    // result for a plan.
+    private static final String AppendToResultTag = "appendToResult".toLowerCase(Locale.ROOT);
+
+    /**
+     * Convert a plan xml string to a plan
+     *
+     * @param xmlString The plan xml string
+     * @param goal The goal for the plan
+     * @param context The semantic kernel context
+     * @return The plan
+     * @throws PlanningException
+     */
+    public static Plan toPlanFromXml(String xmlString, String goal, SKContext context)
+            throws PlanningException {
+
+        try {
```

Comment:
This looks like a good candidate to be broken up.

@lemillermicrosoft this copies the implementation in .Net but I think it would be easier to read if we broke this up e.g. had helper methods like `handleTextNode`, `isFunctionNode()`, `handleFunctionNode()`, ...

What do you think?

Reply:
Not concerns from me :)

Reply:
I have cleaned this up a bit, let me know what you think